File: 6º Workshop - Expressão Regular/seminario.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

import re
from optparse import OptionParser

def blob():
    return open('pendrive_filesystem.dump', 'r').read()

def search(pattern):

    """ Seminario de seguranca
        Metodo para pesquisa do padrao regex
    """

    pattern = re.compile(r'{}'.format(pattern))
    print(pattern.findall(blob()))


if __name__ == '__main__':

    parser = OptionParser()
    usage = "usage: %prog [options] arg1"
    parser.add_option("-s", "--search", type="string",
        help="-s string, --search string",
        dest="pattern", default=".*")
    (options, args) = parser.parse_args()

    search(options.pattern)

File: 6º Workshop - Expressão Regular/seminario.pl
#!/usr/bin/perl
use strict;
use warnings;
use Getopt::Long qw(GetOptions);
use Path::Tiny;
use autodie;

my $dir = path("./");

my $file = $dir->child("pendrive_filesystem.dump");

sub blob {
    return $file->openr_utf8();
}

sub search {
    my $pattern = $_[0];
    my $content = blob();
    while( my $line = $content->getline() ) {
       $line =~ /$pattern/ and print $line;
    }
}

my $pattern;
GetOptions('search=s' => \$pattern) or die "Usage: $0 --search PATTERN\n";
if($pattern) {
    search($pattern)
}